libqtile/backend/x11/core.py

```diff
@@ -600,25 +601,60 @@ def handle_ClientMessage(self, event) -> None:  # noqa: N802
             except IndexError:
                 logger.debug("Invalid desktop index: %s" % index)
 
+    def fake_KeyPress(self, keycode, needs_release=True) -> None:
+        xtest = self.conn.conn(xcffib.xtest.key)
+        # The key needs to be released as it was previously pressed
+        if needs_release:
+            xtest.FakeInput(3, keycode, xcffib.xproto.Time.CurrentTime, 0, 0, 0, 0)
+            self.flush()
+        # Press the key
+        xtest.FakeInput(2, keycode, xcffib.xproto.Time.CurrentTime, 0, 0, 0, 0)
+        self.flush()
+        # Release the key
+        xtest.FakeInput(3, keycode, xcffib.xproto.Time.CurrentTime, 0, 0, 0, 0)
+        self.flush()
+
     def handle_KeyPress(self, event) -> None:  # noqa: N802
         assert self.qtile is not None
 
         keysym = self.conn.code_to_syms[event.detail][0]
-        self.qtile.process_key_event(keysym, event.state & self._valid_mask)
+        key, handled = self.qtile.process_key_event(keysym, event.state & self._valid_mask)
+
+        # As we're grabbing async we can't just replay it, so...
+        # We need to forward the event to the focused window
+        if not handled and key:
+            # We need to ungrab the key as otherwise we get an event loop
+            self.ungrab_key(key)
+            # Modifier is already pressed, just repeat the keycode
```

Comment:
This is not the case when it is send from `simulate_keypress`, handle this or leave this to external tools such as xdotool? If we want to leave it to xdotool I would need to add a `simulated` argument again (with default `False`)

Reply:
Yeah, i think leaving that simulated keypress code the way it was is fine

Reply:
I have done a middle ground, I have made simulate keypress closer to an x11 event so that we can in the future maybe support this. But for now simulated keypresses will only elicit qtile functions through the bindings, but will not actually forward it to a window. Xdotool is fine for this use case